platform/linux-generic/odp_schedule_basic.c

```diff
@@ -720,9 +727,32 @@ static inline int poll_pktin(uint32_t qi)
 
 		if (num_pktin == 0)
 			sched_cb_pktio_stop_finalize(pktio_index);
+
+		return num;
 	}
 
-	return num;
+	if (atomic) {
```

Comment:
I wanted to be conservative and not change synchronization of parallel queues yet. I'll do another patch on top, so that it's easy to undo parallel optimization later if necessary.

Reply:
That makes sense.